vm/address/src/lib.rs

```diff
@@ -3,408 +3,29 @@
 
 mod errors;
 mod network;
-mod payload;
-mod protocol;
 pub use self::errors::Error;
 pub use self::network::Network;
-pub use self::payload::{BLSPublicKey, Payload};
-pub use self::protocol::Protocol;
 
-use data_encoding::Encoding;
+// use data_encoding::Encoding;
 #[allow(unused_imports)]
 use data_encoding_macro::{internal_new_encoding, new_encoding};
-use encoding::{blake2b_variable, serde_bytes, Cbor};
 use once_cell::sync::OnceCell;
-use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
-use std::hash::Hash;
-use std::str::FromStr;
-use std::{borrow::Cow, fmt};
 
-/// defines the encoder for base32 encoding with the provided string with no padding
-const ADDRESS_ENCODER: Encoding = new_encoding! {
-    symbols: "abcdefghijklmnopqrstuvwxyz234567",
-    padding: None,
+pub use fvm_shared::address::{
+    checksum, validate_checksum, Address, BLSPublicKey, Payload, Protocol, BLS_PUB_LEN,
+    CHECKSUM_HASH_LEN, PAYLOAD_HASH_LEN, SECP_PUB_LEN,
 };
 
-/// Hash length of payload for Secp and Actor addresses.
-pub const PAYLOAD_HASH_LEN: usize = 20;
-
-/// Uncompressed secp public key used for validation of Secp addresses.
-pub const SECP_PUB_LEN: usize = 65;
-
-/// BLS public key length used for validation of BLS addresses.
-pub const BLS_PUB_LEN: usize = 48;
-
-lazy_static::lazy_static! {
-    static ref BLS_ZERO_ADDR_BYTES: BLSPublicKey = {
-        let bz_addr = Address::from_str("f3yaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaby2smx7a");
-        if let Ok(Address {payload: Payload::BLS(pubkey), ..}) = bz_addr {
-            pubkey
-        } else {
-            panic!("failed to parse BLS address from provided BLS_ZERO_ADDR string")
-        }
-    };
-}
-
-/// Length of the checksum hash for string encodings.
-pub const CHECKSUM_HASH_LEN: usize = 4;
-
-const MAX_ADDRESS_LEN: usize = 84 + 2;
-const MAINNET_PREFIX: &str = "f";
-const TESTNET_PREFIX: &str = "t";
-
-#[cfg(feature = "json")]
-const UNDEF_ADDR_STRING: &str = "<empty>";
-
 // TODO pull network from config (probably)
 pub static NETWORK_DEFAULT: OnceCell<Network> = OnceCell::new();
 
-/// Address is the struct that defines the protocol and data payload conversion from either
-/// a public key or value
-#[derive(PartialEq, Eq, Clone, Debug, Hash, Copy)]
-pub struct Address {
```

Comment:
why is all this removed? was it moved or unused?

Reply:
The data structure has moved to the `fvm_shared` crate.